compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3255,19 +3255,24 @@ TR::Register *OMR::X86::TreeEvaluator::arraytranslateEvaluator(TR::Node *node, T
    TR::Register *srcPtrReg, *dstPtrReg, *transTableReg, *termCharReg, *lengthReg;
    bool stopUsingCopyReg1 = TR::TreeEvaluator::stopUsingCopyRegAddr(node->getChild(0), srcPtrReg, cg);
    bool stopUsingCopyReg2 = TR::TreeEvaluator::stopUsingCopyRegAddr(node->getChild(1), dstPtrReg, cg);
-   bool stopUsingCopyReg4 = TR::TreeEvaluator::stopUsingCopyRegInteger(node->getChild(3), termCharReg, cg);
+   bool stopUsingCopyReg4 = false;
+   if (!sourceByte)
+      {
+      stopUsingCopyReg4 = TR::TreeEvaluator::stopUsingCopyRegInteger(node->getChild(3), termCharReg, cg);
+      }
```

Comment:
Normally, an opcode evaluator needs to evaluate and decrement the reference count on all its children.  I've seen many problems over the years where this wasn't followed and one problem or another was introduced.  Exceptions to this rule can (and have) been made in the past in controlled circumstances, and this opcode may very well be one of those.

But I'm seeing the code for this evaluator for the first time and I don't have deep familiarity with the opcode.  If you're confirming that skipping the evaluation of the 3rd child is correct in this case then I'm fine with your assessment.

I'm troubled by the implementation of this opcode, but that's not your problem... :-)